### PR TITLE
Add fallback for missing error.html

### PR DESCRIPTION
**Pull request by fiete201**

### CODE REVIEW

**Comment by fiete201** [comment]
https://github.com/The-Compiler/qutebrowser/issues/1466

File: qutebrowser/utils/jinja.py [hunk 1]
```diff
@@ -31,6 +31,49 @@
...
+    <body>
+        <div id="error-container">
+	<table>
+		<tr>
+			<td style="width: 10%; vertical-align: top;">
+				<img style="width: 100%; display: block; max-width: 256px;" src="{{ data_url("img/broken_qutebrowser_logo.png") }}" />
+			</td>
+			<td style="padding-left: 40px;">
+        <p><span style="font-size:120%;color:red">The error.html template could not be found!<br>Please check your qutebrowser installation</span><br>'''
+html_fallback2 = '''</p>
```
Comment (The-Compiler) [inline qutebrowser/utils/jinja.py]: Might be easier to simply use one string with something like `%ERROR%` and use `.replace()` to format that in.

File: qutebrowser/utils/jinja.py [hunk 2]
```diff
@@ -47,8 +90,8 @@ def get_source(self, _env, template):
         path = os.path.join(self._subdir, template)
         try:
             source = utils.read_file(path)
-        except OSError:
-            raise jinja2.TemplateNotFound(template)
+        except OSError as e:
+            source = html_fallback + str(e) + html_fallback2;
```
Comment (The-Compiler) [inline qutebrowser/utils/jinja.py]: Please use `html.escape(str(e))` to have proper HTML escaping (with the `html` module in the stdlib)

File: qutebrowser/utils/jinja.py [hunk 3]
```diff
@@ -31,6 +31,49 @@
 
 from PyQt5.QtCore import QUrl
 
+html_fallback = '''<!DOCTYPE html>
```
Comment (The-Compiler) [inline qutebrowser/utils/jinja.py]: nitpick: please use `"""` here

File: qutebrowser/utils/jinja.py [hunk 4]
```diff
@@ -31,6 +32,49 @@
...
+        <div id="error-container">
+	<table>
+		<tr>
+			<td style="width: 10%; vertical-align: top;">
+				<img style="width: 100%; display: block; max-width: 256px;" src="{{ data_url("img/broken_qutebrowser_logo.png") }}" />
+			</td>
+			<td style="padding-left: 40px;">
+        <p><span style="font-size:120%;color:red">The error.html template could not be found!<br>Please check your qutebrowser installation</span><br>
+%ERROR%</p>
+				<h1>Unable to load page</h1>
```
Comment (The-Compiler) [inline qutebrowser/utils/jinja.py]: Error pages also are shown for various other errors, so I'm not sure if we really should show this here - what about simply showing the meta-error and nothing else?

File: qutebrowser/utils/jinja.py [hunk 5]
```diff
@@ -47,8 +91,9 @@ def get_source(self, _env, template):
         path = os.path.join(self._subdir, template)
         try:
             source = utils.read_file(path)
-        except OSError:
-            raise jinja2.TemplateNotFound(template)
+        except OSError as e:
+            source = html_fallback.replace("%ERROR%", html.escape(str(e)))
+            log.misc.error("The error.html template could not be found" + path)
```
Comment (The-Compiler) [inline qutebrowser/utils/jinja.py]: The ` + path` seems a bit weird here - shouldn't this be `"could not be found at " + path` or so?

File: qutebrowser/utils/jinja.py (hunk 4 above)
Comment (fiete201) [inline qutebrowser/utils/jinja.py]: yes removed that and only left the general meta error

File: qutebrowser/utils/jinja.py (hunk 5 above)
Comment (fiete201) [inline qutebrowser/utils/jinja.py]: Also fixed that. But for some reason the logged message is printed twice

**Pull request merged**